### PR TITLE
Add custom menu

### LINKED ISSUE #583: Add a custom menu
* **I'm submitting a ...**
  - [ ] bug report
  - [x] feature request

* **What is the current status?**
Menu bar consists of irrelevant menu items. Specially the _help_ menu. It includes the documentation and other links to electronjs. Instead, there should be links to PSLab documentation.
fjaskldjfa

* **What is the expected behavior?**
Remove irrelevant options and add help for **PSLab Documentation** 

* **Would you like to work on it?**
Yes

### PR DESCRIPTION
**Pull request by MuhammadWasif**
Fix #583 
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [x] Bug fix
- [ ] Feature implementation
- [ ] Doc updates

* **What changes have you introduced?**
Relevant menu items. Now, contains documentation for PSLab sites

* **Preview / Steps to verify your work**:
Explore the menu bar
fka

### CODE REVIEW

File: public/electron.js [hunk 1]
```diff
@@ -17,7 +17,9 @@ const loadBalancer = require('electron-load-balancer');
 
 const { app } = electron;
 const { BrowserWindow } = electron;
+const { Menu } = electron;
```
Comment (Sid911) [inline public/electron.js]: Please use one single statement for importing from 1 package if its not necessary 😇, Menu app and Browser window in this case

Comment (MuhammadWasif) [reply public/electron.js]: `app` and `BrowserWindow` were already imported separately. So, I thought `Menu` should be imported separately too  😅

Comment (Sid911) [reply public/electron.js]: yeah but there no need when you can use { app, BrowserWindow, Menu} ... it just keeps the code tidier :)

Comment (MuhammadWasif) [reply public/electron.js]: Fixed...! ✔️

**Comment by MuhammadWasif** [comment]
@cweitat Please review...! All things are working fine. It would be great if you merge this  🙂

**Comment by cweitat** [comment]
@MuhammadWasif we need two approvals before we can merge; our code of practice

**Comment by MuhammadWasif** [comment]
@fcartegnie Unfortunately, I can't squash commits now due to some problem. I would take care next time. It would be nice if you approve this time as it's not causing major problem. Thanks   😊

**Comment by fcartegnie** [comment]
> @fcartegnie Unfortunately, I can't squash commits now due to some problem. I would take care next time. It would be nice if you approve this time as it's not causing major problem. Thanks blush

And I don't see here a situation that would need to merge commits in a way that would be rejected elsewhere, bending the rules. That can't be merged unless the changes are unitary and related to the merge feature.

**Comment by MuhammadWasif** [comment]
@cweitat Looks like I've got two approvals. Please merge it now  🥺 🥺 🥺

**Comment by cweitat** [comment]
@MuhammadWasif two approvals from core.

**Comment by Sid911** [comment]
@MuhammadWasif you can try using statements
`git reset --soft HEAD~2` then
`git commit -m "your message"`
`git push --force`

**Comment by adityastic** [comment]
@cweitat @CloudyPadmal are you sure we should have this feature? I'm not sure about having a menu bar, doesn't look aesthetically proper nor it serves any purpose. We should rather have it in our drawer, somewhere below settings. The purpose of a menu bar is to provide options that will come in use regularly, I'm not sure if a complete menu bar is needed to serve a purpose of providing docs.

**Comment by Sid911** [comment]
@adiyastic yeah... I always thought of that but it was already there before this... So I thought it must be there for a reason 🤔..Personally I would implement a frameless window and add a custom app bar with close controls etc. For consistency across the platform.

**Comment by Sid911** [comment]
Although I am out of station for 1 more day.. I could take up the issue after 2 days... As I already have a functionality good looking working component of app bar.. it's just about chucking that in... And changing colors..😇

**Comment by Sid911** [comment]
Can just anyone tell me if this application uses Redux for centralized storage and does it use persisting state in any way?

**Comment by MuhammadWasif** [comment]
@fcartegnie Commits have been converted it into 1 single commit. I hope you'll approve now

**Comment by MuhammadWasif** [comment]
> @cweitat @CloudyPadmal @fcartegnie are you sure we should have this feature? I'm not sure about having a menu bar, doesn't look aesthetically proper nor it serves any purpose. We should rather have it in our drawer, somewhere below settings. The purpose of a menu bar is to provide options that will come in use regularly, I'm not sure if a complete menu bar is needed to serve a purpose of providing docs.

I think it's better than a menu that contains links for electron docs etc

**Comment by Sid911** [comment]
yeah its better @MuhammadWasif  than before for sure..

**Comment by MuhammadWasif** [comment]
> Add support for macos, the menu in macos is different. It should consider all the platforms

@adityastic 
I'm not using Mac. Docs says the menu works for all operating systems. Can you please provide a screenshot of what it looks like in Mac?

**Comment by adityastic** [comment]
https://electronjs.org/docs/api/menu, this should help

**Comment by MuhammadWasif** [comment]
@adityastic The changes I made were the only deficiency for macOS support. All other stuff seems to work fine according to docs. I hope you will approve it now

**Comment by MuhammadWasif** [comment]
@cweitat @fcartegnie Waiting for your reviews

**Comment by MuhammadWasif** [comment]
@mariobehling Okay, no problem. But it would be better if someone would have told me earlier  😓 😓
It's quite late now

**Pull request closed without merging**